File: backend/backend/Areas/Communication/Models/UserMeeting.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Areas.Identity.Models;

namespace backend.Areas.Communication.Models;

public class UserMeeting
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    public string UserId { get; set; }
    [ForeignKey(nameof(UserId))]
    public User User { get; set; }

    public int MeetingId { get; set; }
    [ForeignKey(nameof(MeetingId))]
    public Meeting Meeting { get; set; }
}

File: frontend/lib/models/user_meeting.dart
import 'package:frontend/models/User.dart';
import 'package:frontend/models/meeting.dart';

class UserMeeting {
  int? id;
  String? userId;
  User? user;
  int? meetingId;
  Meeting? meeting;
  UserMeeting({
    this.id,
    this.userId,
    this.user,
    this.meetingId,
    this.meeting
  });

  factory UserMeeting.fromJson(Map<String, dynamic> json) {
    return UserMeeting(
      id: json['id'],
      userId: json['userId'],
      meetingId: json['meetingId'],
      meeting: json['meeting'],
      user: json['user'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'meetingId': meetingId,
      'meeting': meeting,
      'user': user,
    };
  }
}